okta_request_condition 409 errors

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
        Before submitting a bug report, we ask that you first [search existing issues](https://github.com/okta/terraform-provider-okta/issues) and [pull requests](https://github.com/okta/terraform-provider-okta/pulls) to see if someone else may have experienced the same issue or may have already submitted a fix for it. This helps to keep all relevant information in one place, including any potential workarounds.

### A Note on Terraform Core Issues

We also ask that you consider whether your issue may be related to Terraform Core. If you are running into one of the following scenarios, we recommend [opening an issue](https://github.com/hashicorp/terraform/issues/new/choose) in the Terraform Core repository instead:

* [Configuration Language](https://developer.hashicorp.com/terraform/language) or resource ordering issues
* [State](https://developer.hashicorp.com/terraform/language/state) and [State Backend](https://developer.hashicorp.com/terraform/language/backend) issues
* [Provisioner](https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax) issues
* [Registry](https://registry.terraform.io/) issues
* Issues that span resources across multiple providers


### Terraform Version & Okta Provider Version(s)

```
OpenTofu v1.11.1
on darwin_arm64
+ provider registry.opentofu.org/okta/okta v6.5.1
```

### Affected Resource(s)

okta_request_condition

### Can this be done in the Admin UI?

No

### Can this be done in the actual API call?

Unsure

### Customer Information

Organization Name: hh-wf
Paid Customer: yes


### Terraform Configuration

```tf
resource "okta_request_condition" "this" {
  resource_id          = var.app_id
  approval_sequence_id = var.approval_sequence_id
  name                 = var.name
  description          = var.description

  access_scope_settings {
    type = "GROUPS"
    dynamic "ids" {
      for_each = var.assignment_groups
      content {
        id = ids.value
      }
    }
  }

  requester_settings {
    type = "GROUPS"
    dynamic "ids" {
      for_each = var.allowed_requesters
      content {
        id = ids.value
      }
    }
  }

  access_duration_settings {
    type     = "ADMIN_FIXED_DURATION"
    duration = "PT${var.duration_hours}H"
  }

```


### Debug Output

```shell
  # module.write_requests["0"].okta_request_condition.this will be updated in-place
  ~ resource "okta_request_condition" "this" {
      ~ created              = "2025-12-02T19:35:30Z" -> (known after apply)
      ~ created_by           = "pcabotwfn9jQEgCwi697" -> (known after apply)
      ~ id                   = "rcorq8h42VPbLkyzB696" -> (known after apply)
      ~ last_updated         = "2026-01-21T14:05:53Z" -> (known after apply)
      ~ last_updated_by      = "pcabotwfn9jQEgCwi697" -> (known after apply)
        name                 = "Write Access 0"
      ~ priority             = 2 -> (known after apply)
      ~ status               = "ACTIVE" -> (known after apply)
        # (3 unchanged attributes hidden)

      ~ access_scope_settings {
            # (1 unchanged attribute hidden)

          ~ ids {
              ~ id = "00gzeej5ulIeS3oMY697" -> "00gwmo4lr9TbY4RB3697"
            }
          ~ ids {
              ~ id = "00gz3zlpf6PCdrgo7697" -> "00gzeej5ulIeS3oMY697"
            }
          ~ ids {
              ~ id = "00gz3zgyzo99Az0X7697" -> "00gz3zlpf6PCdrgo7697"
            }
          ~ ids {
              ~ id = "00gwmo7xo9TMZyhTS697" -> "00gz3zgyzo99Az0X7697"
            }
          ~ ids {
              ~ id = "00gy8slf73DoQ3Gup697" -> "00gwmo7xo9TMZyhTS697"
            }
          ~ ids {
              ~ id = "00gy8sq4uxdObBxPm697" -> "00gy8slf73DoQ3Gup697"
            }
          ~ ids {
              ~ id = "00gx83ns0uqI01CsJ697" -> "00gy8sq4uxdObBxPm697"
            }
          ~ ids {
              ~ id = "00gwmo615aGg1TL2w697" -> "00gx83ns0uqI01CsJ697"
            }
          ~ ids {
              ~ id = "00gzaxa9n810em9vn697" -> "00gwmo615aGg1TL2w697"
            }
          + ids {
              + id = "00gzaxa9n810em9vn697"
            }

            # (14 unchanged blocks hidden)
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 3 to change, 0 to destroy.

─────────────────────────────────────────────────────────────────────────────

module.read_requests["0"].okta_request_condition.this: Modifying... [id=rcorq7yi4Z0Pcvebj696]
module.write_requests["0"].okta_request_condition.this: Modifying... [id=rcorq8h42VPbLkyzB696]
module.write_requests["0"].okta_request_condition.this: Modifications complete after 0s [id=rcorq8h42VPbLkyzB696]

╷
│ Error: Error updating Request conditions
│ 
│   with module.read_requests["0"].okta_request_condition.this,
│   on modules/access-requests/main.tf line 1, in resource "okta_request_condition" "this":
│    1: resource "okta_request_condition" "this" {
│ 
│ Could not update Request conditions, unexpected error: 409 Conflict
╵
Error: OpenTofu exited with code 1.
Error: Process completed with exit code 1.
```

### Expected Behavior

Changes in `okta_request_condition.access_scope_settings` be updated in the request condition.

### Actual Behavior

The provider receives a 409 from Okta and then returns an error. I had a ticket opened with Okta enterprise support on this issue. They assured that this was due to the provider making calls to the API in rapid succession while the resource was in the process of being updated by the previous API call. This causes the 409 error. Oddly enough, when this happens it seems that the resource in Okta is actually updated as the next TF run shows no diff. The reindexing of id's seems to be largely to blame for the API getting overrun with requests.

### Steps to reproduce

1. `terraform apply`

### Important Factoids

_No response_

### References

_No response_

--- a/okta/services/governance/resource_request_condition.go
+++ b/okta/services/governance/resource_request_condition.go
@@ -2,6 +2,8 @@ package governance
 
 import (
 	"context"
+	"github.com/okta/terraform-provider-okta/okta/api"
+	"net/http"
 	"strings"
 	"time"
 
@@ -193,7 +195,7 @@ func (r *requestConditionResource) Create(ctx context.Context, req resource.Crea
 	if resp.Diagnostics.HasError() {
 		return
 	}
-
+	ctx = context.WithValue(ctx, api.RetryOnStatusCodes, []int{http.StatusConflict})
 	requestConditionResp, _, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.CreateResourceRequestConditionV2(ctx, data.ResourceId.ValueString()).RequestConditionCreatable(createRequestCondition(data)).Execute()
 	if err != nil {
 		resp.Diagnostics.AddError(
@@ -238,8 +240,14 @@ func (r *requestConditionResource) Read(ctx context.Context, req resource.ReadRe
 	}
 
 	// Read API call logic
-	readRequestConditionResp, _, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.GetResourceRequestConditionV2(ctx, data.ResourceId.ValueString(), data.Id.ValueString()).Execute()
+	readRequestConditionResp, httpResp, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.GetResourceRequestConditionV2(ctx, data.ResourceId.ValueString(), data.Id.ValueString()).Execute()
 	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == http.StatusNotFound {
+			// Resource was deleted outside Terraform — remove from state so
+			// the next plan recreates it rather than erroring on every refresh.
+			resp.State.RemoveResource(ctx)
+			return
+		}
 		resp.Diagnostics.AddError(
 			"Error reading Request conditions",
 			"Could not read Request conditions, unexpected error: "+err.Error(),
@@ -270,6 +278,7 @@ func (r *requestConditionResource) Update(ctx context.Context, req resource.Upda
 	}
 
 	// Update API call logic
+	ctx = context.WithValue(ctx, api.RetryOnStatusCodes, []int{http.StatusConflict})
 	updatedRequestCondition, _, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.UpdateResourceRequestConditionV2(ctx, data.ResourceId.ValueString(), state.Id.ValueString()).RequestConditionPatchable(createRequestConditionPatch(data)).Execute()
 	if err != nil {
 		resp.Diagnostics.AddError(
@@ -354,8 +363,21 @@ func (r *requestConditionResource) Delete(ctx context.Context, req resource.Dele
 	}
 
 	// Delete API call logic
-	_, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.DeleteResourceRequestConditionV2(ctx, data.ResourceId.ValueString(), state.Id.ValueString()).Execute()
+	deleteResp, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.DeleteResourceRequestConditionV2(ctx, data.ResourceId.ValueString(), state.Id.ValueString()).Execute()
 	if err != nil {
+		if deleteResp != nil && deleteResp.StatusCode == http.StatusNotFound {
+			// Already deleted outside Terraform — treat as success.
+			return
+		}
+		if deleteResp != nil && deleteResp.StatusCode == http.StatusInternalServerError {
+			// The governance API sometimes returns 500 even when the delete
+			// succeeded. Verify by attempting a GET — if the resource is gone
+			// (404) we can safely treat the delete as successful.
+			_, verifyResp, _ := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.GetResourceRequestConditionV2(ctx, data.ResourceId.ValueString(), state.Id.ValueString()).Execute()
+			if verifyResp != nil && verifyResp.StatusCode == http.StatusNotFound {
+				return
+			}
+		}
 		resp.Diagnostics.AddError(
 			"Error deleting Request conditions",
 			"Could not delete Request conditions, unexpected error: "+err.Error(),

--- a/okta/services/governance/resource_request_condition.go
+++ b/okta/services/governance/resource_request_condition.go
@@ -2,10 +2,11 @@ package governance
 
 import (
 	"context"
-	"github.com/okta/terraform-provider-okta/okta/api"
 	"net/http"
 	"strings"
 	"time"
+
+	"github.com/okta/terraform-provider-okta/okta/api"
 
 	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
 	"github.com/hashicorp/terraform-plugin-framework/diag"
@@ -289,7 +290,7 @@ func (r *requestConditionResource) Update(ctx context.Context, req resource.Upda
 	}
 
 	// Handle status changes
-	if !data.Status.IsNull() && !state.Status.IsNull() {
+	if !data.Status.IsNull() {
 		oldStatus := state.Status.ValueString()
 		newStatus := data.Status.ValueString()
 
@@ -332,16 +333,9 @@ func (r *requestConditionResource) Update(ctx context.Context, req resource.Upda
 }
 
 func (r *requestConditionResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
-	var data, state requestConditionResourceModel
+	var state requestConditionResourceModel
 
 	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
-	if resp.Diagnostics.HasError() {
-		return
-	}
-
-	// Read Terraform prior state Data into the model
-	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
-
 	if resp.Diagnostics.HasError() {
 		return
 	}
@@ -363,7 +357,7 @@ func (r *requestConditionResource) Delete(ctx context.Context, req resource.Dele
 	}
 
 	// Delete API call logic
-	deleteResp, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.DeleteResourceRequestConditionV2(ctx, data.ResourceId.ValueString(), state.Id.ValueString()).Execute()
+	deleteResp, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.DeleteResourceRequestConditionV2(ctx, state.ResourceId.ValueString(), state.Id.ValueString()).Execute()
 	if err != nil {
 		if deleteResp != nil && deleteResp.StatusCode == http.StatusNotFound {
 			// Already deleted outside Terraform — treat as success.
@@ -373,7 +367,7 @@ func (r *requestConditionResource) Delete(ctx context.Context, req resource.Dele
 			// The governance API sometimes returns 500 even when the delete
 			// succeeded. Verify by attempting a GET — if the resource is gone
 			// (404) we can safely treat the delete as successful.
-			_, verifyResp, _ := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.GetResourceRequestConditionV2(ctx, data.ResourceId.ValueString(), state.Id.ValueString()).Execute()
+			_, verifyResp, _ := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.GetResourceRequestConditionV2(ctx, state.ResourceId.ValueString(), state.Id.ValueString()).Execute()
 			if verifyResp != nil && verifyResp.StatusCode == http.StatusNotFound {
 				return
 			}
@@ -575,7 +569,6 @@ func createRequestConditionPatch(data requestConditionResourceModel) governance.
 	if !data.Priority.IsNull() {
 		patch.Priority = data.Priority.ValueInt32Pointer()
 	}
-	patch.ApprovalSequenceId = data.ApprovalSequenceId.ValueStringPointer()
 	var accessScopeSettings governance.AccessScopeSettingsCreatableAccessScopeSettings
 	if data.AccessScopeSettings.Type.ValueString() == "GROUPS" {
 		if accessScopeSettings.AccessScopeSettingsCreatableGroupAccessScopeSettings == nil {

